observer.py

```diff
@@ -0,0 +1,17 @@
+class Observer:
+    def __init__(self) -> None:
+        self.action = 0
+        self.infos: list[int] = []
+
+    def alert(self, action: int) -> None:
+        self.action = action
+
+    def act(self, manager) -> None:
```

Comment:
Annotate. It is very inconsistent. Sometimes there is a type, sometimes there isn't. `manager: ...`